### PR TITLE
Added parent prefix

### PR DESCRIPTION
**Pull request by mikiquantum**

### CODE REVIEW

File: examples/documents/documents.go [hunk 1]
```diff
@@ -3,8 +3,9 @@ package documentspb
 //go:generate protoc -I $PROTOBUF/src/ -I. -I $GOPATH/src --go_out=$GOPATH/src/ examples.proto
 
 import (
-	"github.com/golang/protobuf/ptypes"
 	"time"
+
+	"github.com/golang/protobuf/ptypes"
```
Comment (vedhavyas) [inline examples/documents/documents.go]: Nice!

File: proofs/tree.go [hunk 2]
```diff
@@ -747,14 +753,24 @@ func newMessageFlattener(message, messageSalts proto.Message, saltsLengthSuffix
 	return &f
 }
 
+func ensurePrefixTrailingDot(prefix string) string {
+	out := prefix
+	if out != "" {
+		if isDot := out[len(out)-1]; isDot != '.' {
+			out = out + "."
+		}
+	}
```
Comment (vedhavyas) [inline proofs/tree.go]:
```
	if !strings.HasSuffix(prefix, ".") {
		prefix = prefix + "."
	}
```

This should handle all the edge cases IMO. What do you think?

File: examples/simple.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ import (
 	"crypto/sha256"
 	"encoding/json"
 	"fmt"
+
```
Comment (vedhavyas) [inline examples/simple.go]: Love it!

File: proofs/tree.go (hunk 2 above)
Comment (mikiquantum) [reply proofs/tree.go]: ok, I just need to add only when prefix is not empty. Pushing now.

**Pull request merged**